sherpa-onnx/csrc/onnxrt-execution-provider-config.h

```diff
@@ -0,0 +1,94 @@
+// sherpa-onnx/csrc/online-transducer-model-config.h
+//
+// Copyright (c)  2023  Xiaomi Corporation
+#ifndef SHERPA_ONNX_CSRC_ONNXRT_EXECUTION_PROVIDER_CONFIG_H_
+#define SHERPA_ONNX_CSRC_ONNXRT_EXECUTION_PROVIDER_CONFIG_H_
+
+#include <string>
+
+#include "sherpa-onnx/csrc/parse-options.h"
+#include "onnxruntime_cxx_api.h"  // NOLINT
+
+namespace sherpa_onnx {
+
+struct OnnxrtCudaConfig {
+  int32_t device = 0;
+  int32_t cudnn_conv_algo_search = OrtCudnnConvAlgoSearchHeuristic;
+
+  OnnxrtCudaConfig() = default;
+  OnnxrtCudaConfig(const int32_t &device,
+                   const int32_t &cudnn_conv_algo_search
+                   )
+      : device(device), cudnn_conv_algo_search(cudnn_conv_algo_search) {}
+
+  void Register(ParseOptions *po);
+  bool Validate() const;
+
+  std::string ToString() const;
+};
+
+struct OnnxrtTensorrtConfig {
+  uint32_t device_id = 0;
+  uint32_t trt_max_workspace_size = 2147483648;
+  uint32_t trt_max_partition_iterations = 10;
+  uint32_t trt_min_subgraph_size = 5;
+  uint32_t trt_fp16_enable = 1;
+  uint32_t trt_detailed_build_log = 0;
+  uint32_t trt_engine_cache_enable = 1;
+  std::string trt_engine_cache_path = ".";
+  uint32_t trt_timing_cache_enable = 1;
+  std::string trt_timing_cache_path = ".";
+
+  std::string encoder;
+  std::string decoder;
+  std::string joiner;
```

Comment:
Could you remove model-related fields from this config?

This config can be used for different types of models, not limited to transducer models.

Reply:
yess yes, it is a very early patch, and just created templates.
will remove them, will request for review once completed

Reply:
Thanks!